vector/src/main/java/im/vector/home/NewHomeFragment.kt

```diff
@@ -0,0 +1,185 @@
+package im.vector.home
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.content.ContextCompat
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentManager
+import androidx.fragment.app.FragmentPagerAdapter
+import im.vector.R
+import im.vector.adapters.HomeRoomAdapter
+import im.vector.fragments.AbsHomeFragment
+import im.vector.ui.themes.ThemeUtils.getColor
+import im.vector.ui.themes.ThemeUtils.setTabLayoutTheme
+import im.vector.util.HomeRoomsViewModel
+import im.vector.util.PreferencesManager
+import im.vector.util.RoomUtils
+import kotlinx.android.synthetic.main.fragment_view_pager_tab.*
+import org.matrix.androidsdk.data.Room
+import org.matrix.androidsdk.data.RoomTag
+
+class NewHomeFragment : AbsHomeFragment(), HomeRoomAdapter.OnSelectRoomListener, RegisterListener {
+    val dataUpdateListeners = ArrayList<UpDateListener>()
+    var result: HomeRoomsViewModel.Result? = null
+
+    override fun getLayoutResId(): Int {
+        return R.layout.fragment_view_pager_tab
+    }
+
+    override fun onCreateView(
+            inflater: LayoutInflater, container: ViewGroup?,
+            savedInstanceState: Bundle?
+    ): View? {
+        return inflater.inflate(getLayoutResId(), container, false)
+    }
+
+    override fun onActivityCreated(savedInstanceState: Bundle?) {
+        super.onActivityCreated(savedInstanceState)
+        pager.offscreenPageLimit = 3
+        pager.adapter = HomePagerAdapter(childFragmentManager, resources.getStringArray(R.array.home_tabs))
+        tabLayout.setupWithViewPager(pager)
+        activity?.let { activity ->
+            setTabLayoutTheme(activity, tabLayout)
+            mPrimaryColor = getColor(activity, R.attr.vctr_tab_home)
+            mSecondaryColor = getColor(activity, R.attr.vctr_tab_home_secondary)
+            mFabColor = ContextCompat.getColor(activity, R.color.tab_rooms)
+            mFabPressedColor = ContextCompat.getColor(activity, R.color.tab_rooms_secondary)
+        }
+    }
+
+    override fun getRooms(): MutableList<Room> {
```

Comment:
This function is also in BaseNewHomeIndividualFragment, could this class subclass BaseNewHomeIndividualFragment, or could the code be centralised in another way?

Reply:
as the previous comment mentions, this is from the Riot base view class. Some of the variables are easily accessible by extending that base class such as session which has most of the data.

Reply:
CommunicateHomeFragment does not extend BaseCommunicateHomeIndividualFragment. individual fragments (favorite, invite, low priority and normal) extends BaseCommunicateHomeIndividualFragment.

Reply:
"CommunicateHomeFragment does not extend BaseCommunicateHomeIndividualFragment" correct I am suggesting that CommunicateHomeFragment extends BaseCommunicateHomeFragment and
individual fragments extends BaseCommunicateHomeIndividualFragment extends BaseCommunicateHomeFragment
with the common code in CommunicateHomeFragment and BaseCommunicateHomeIndividualFragment centralised into BaseCommunicateHomeFragment

Reply:
Basically, getRooms, onFilter, onResetFilter and onRoomResultUpdated are all from abstract class named AbsHomeFragment. Although getrooms function has the same functionality, the other three have separate codes.

Reply:
Yep just getRooms would then be in the BaseCommunicateHomeFragment

Reply:
done.